.github/workflows/ci_release.yml

```diff
@@ -0,0 +1,174 @@
+            packages/dashboard/Dockerfile: packages/dashboard/Dockerfile
+            packages/dnslink-api/Dockerfile: packages/dnslink-api/Dockerfile
+            packages/handshake-api/Dockerfile: packages/handshake-api/Dockerfile
+            packages/health-check/Dockerfile: packages/health-check/Dockerfile
+            packages/website/Dockerfile: packages/website/Dockerfile
+
+  # Run the hadolint dockefile lint if there are changes to any of the
+  # dockerfiles
+  hadolint:
+    needs: dockerfile-changes
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        dockerfile: ${{ fromJSON(needs.dockerfile-changes.outputs.packages) }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: hadolint/hadolint-action@v2.1.0
+        with:
+          dockerfile: ${{ matrix.dockerfile }}
+
+  # Determine if there are python changes
+  python-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      # Check if the changed status of py is true
+      updates: steps.filter.outputs.py == 'true'
+    steps:
+      # Use paths-filter to determine which paths have changes
+      # https://github.com/dorny/paths-filter
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            py: "**.py"
+
+  # Run the hadolint dockefile lint if there are changes to any of the
+  # dockerfiles
+  python-lint:
+    needs: python-changes
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Run Black python linter
+        if: ${{ needs.python-changes.outputs.updates }}
+        uses: SkynetLabs/.github/.github/actions/python-black-lint@master
+      - name: Run Flake8 python linter
+        if: ${{ needs.python-changes.outputs.updates }}
+        uses: SkynetLabs/.github/.github/actions/python-flake8-lint@master
+
+  # Check if there were any changes since the last tag if this is not a push
+  # event
+  changes:
+    needs: [prettier, eslint, hadolint, python-lint]
+    runs-on: ubuntu-latest
+    outputs:
+      updates: ${{steps.changes.outputs.any == 'true'}}
+    if: ${{ github.event_name != 'push' }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # Required due to the weg Git works, without it this action won't be able to find any or the correct tags
+      - uses: SkynetLabs/.github/.github/actions/changes-since-last-tag@master
+
+  # Make a release if
+  # - there were changes and this is a scheduled job
+  # - This is a manually trigger job, i.e. workflow_dispatch
+  release:
+    needs: changes
+    runs-on: ubuntu-latest
+    if: ${{ (needs.changes.outputs.updates == 'true' && github.event_name == 'schedule') || github.event_name == 'workflow_dispatch' }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Version Release
+        uses: SkynetLabs/.github/.github/actions/version-release@master
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
```

Comment:
I don't believe this is a good idea to tag master every night, we should only tag it when we have meaningful changes, I can see bunch of tags that have "fix typo" in the release notes. We also don't have enough code coverage, both unit and integration, to be certain that the release we're tagging is not broken - that's why before deploying to production we're checking the release manually on dev servers and we're supposed to even improve that process to deploy to dev servers even couple of days early to test the potential release. Additionally, if we tag master every night, any code that we will want to revert from master will be a breaking change and accordingly to semver require major version bump. I believe this will harm our processes more that it will help them, my stand is that making a release should be triggered by a human.